### DISCUSSION TITLE
Error in compilation using remote desktop

### DISCUSSION THREAD

**Discussion by bha112** [idaholab/moose#21368] [Q&A Modules: General] [answered]
Hello,

Using this command in my remote desktop for compilation of the input file **mpirun -np 4 ../../home/projects/mastodon/mastodon-opt -i big_model.i ** . Its just printing the same steps 4 times in the terminal, Instead it should use 4 processors for solving the problem. 

Please find the 3 attached Terminal Screenshots , First figure shows that after giving 4 processors command its only Using 
one processors
![mpi_4](https://user-images.githubusercontent.com/107937888/174834102-3f0ce664-3273-47c7-a64e-f43ff6c68577.png)
 , Second and Third figures shows that the same steps is displaying 4 times in the Terminal  
![66](https://user-images.githubusercontent.com/107937888/174834127-f444498a-17be-4c08-ac92-e0dab2dc520e.png)
![55](https://user-images.githubusercontent.com/107937888/174834135-5e867874-d7f3-4b48-8fc6-9ef9b1c691fd.png)

Could you please help me to solve the issue?

Thank you very much!

Bhavesh

**Comment by GiudGiud** [comment, marked as answer]
Hello

I've seen this before when the mpirun does not match the MPI compiler used.
What did you use to compile?
What does
```
which mpirun
echo $CXX
which mpicc
```
return ?

Guillaume

**Reply by bha112** [reply]
Thanks for your response. You were right, I was using a different mpi version (open mpi 4.0.3) than the one that comes with moose tools. Once I activated conda moose environment, mpirun is working fine.

**Reply by milljm** [reply]
This will change in the near future: https://github.com/idaholab/moose/pull/21324

I am removing the dependency of MPICH within moose-tools. Many dependencies will be loosened in this regard. While others will be even tighter (compilers specifically).
